presto-resource-group-managers/src/main/java/com/facebook/presto/resourceGroups/StaticSelector.java

```diff
@@ -50,7 +57,11 @@ public StaticSelector(Optional<Pattern> userRegex, Optional<Pattern> sourceRegex
                 return Optional.empty();
             }
         }
-
+        if (!clientTags.isEmpty()) {
+            if (!context.getTags().containsAll(clientTags)) {
```

Comment:
why `containsAll` instead of an exact match here?

Reply:
A query can potentially have tags for different purposes. For example, it can have a "high_mem" tag, and a "dataswarm" tag, and a "slow_client" tag. We might want to say that as long as it has a "high_mem" tag, we want to put it into the high memory resource group, and we do not care whether it have other tags. Other features (session property manager, for example, might use different tags). Essentially, each tag should describe the request in some way, and when components use tags, they are only looking for inputs with certain characteristics, but not necessarily care about others.